src/Chapter2.hs

```diff
@@ -857,7 +870,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind = go []
+  where
+    go :: [a] -> [a] -> [a]
+    go res [] = res
+    go res (x:xs) = go (x : res) xs
```

Comment:
Bravo! Perfect implementation. And very efficient.

The pattern with `go` helper in function is quite common. We also wrote about it in one of our posts: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go